pkg/workload/workload_test.go

```diff
@@ -643,3 +644,62 @@ func TestResourceUsage(t *testing.T) {
 		})
 	}
 }
+
+func TestAdmissionCheckStrategy(t *testing.T) {
+	cases := map[string]struct {
+		cq                  *kueue.ClusterQueue
+		wl                  *kueue.Workload
+		wantAdmissionChecks sets.Set[string]
+	}{
+		"AdmissionCheckStrategy with a flavor": {
+			wl: utiltesting.MakeWorkload("wl", "ns").
+				ReserveQuota(utiltesting.MakeAdmission("cq").Assignment("cpu", "flavor1", "1").Obj()).
+				Obj(),
+			cq: utiltesting.MakeClusterQueue("cq").
+				AdmissionCheckStrategy(*utiltesting.MakeAdmissionCheckStrategyRule("ac1", "flavor1").Obj()).
+				Obj(),
+			wantAdmissionChecks: sets.New("ac1"),
+		},
+		"AdmissionCheckStrategy without a flavor": {
+			wl: utiltesting.MakeWorkload("wl", "ns").
+				ReserveQuota(utiltesting.MakeAdmission("cq").Assignment("cpu", "flavor1", "1").Obj()).
+				Obj(),
+			cq: utiltesting.MakeClusterQueue("cq").
+				AdmissionCheckStrategy(*utiltesting.MakeAdmissionCheckStrategyRule("ac1").Obj()).
+				Obj(),
+			wantAdmissionChecks: sets.New("ac1"),
+		},
+		"Two AdmissionCheckStrategies, one with flavor, one without flavor": {
+			wl: utiltesting.MakeWorkload("wl", "ns").
+				ReserveQuota(utiltesting.MakeAdmission("cq").Assignment("cpu", "flavor1", "1").Obj()).
+				Obj(),
+			cq: utiltesting.MakeClusterQueue("cq").
+				AdmissionCheckStrategy(
+					*utiltesting.MakeAdmissionCheckStrategyRule("ac1", "flavor1").Obj(),
+					*utiltesting.MakeAdmissionCheckStrategyRule("ac2").Obj()).
+				Obj(),
+			wantAdmissionChecks: sets.New("ac1", "ac2"),
+		},
+		"Workload has no QuotaReserved": {
```

Comment:
Add another case where the admission check doesn't match the flavor of the workload assignment, so that we can check that it's not added to the Workload.

Reply:
Do you mean a different scenario than the `"AdmissionCheckStrategy with an unmatched flavor"`?

Reply:
That sounds like the one I want, but I don't see it in this file. Is it somewhere else?

Reply:
Can you double check if the changes you're looking at are up-to-date?